### PR TITLE
Improve provisioning constant and address feedback.

### PR DESCRIPTION
**Pull request by adamsilverstein**
## Summary

Follow up to https://github.com/google/site-kit-wp/pull/1339, addressing some issues noted in review and QA.

Addresses issue #1212

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: includes/Modules/Analytics.php [hunk 1]
```diff
@@ -1104,7 +1104,7 @@ function ( Google_Service_Analytics_Account $account ) {
 			case 'POST:create-account-ticket':
 				// Cache the create ticket id long enough to verify it upon completion of the terms of service.
 				set_transient(
-					$self::ANALYTICS_PROVISIONING_ACCOUNT_TICKET_ID . '::' . get_current_user_id(),
+					$self::PROVISION_ACCOUNT_TICKET_ID . '::' . get_current_user_id(),
```
Comment (aaemnnosttv) [inline includes/Modules/Analytics.php:1107]: `$self` is still an undefined variable here.
```suggestion
					self::PROVISION_ACCOUNT_TICKET_ID . '::' . get_current_user_id(),
```

**Comment by aaemnnosttv** [comment]
@adamsilverstein there was [one minor fix](https://github.com/google/site-kit-wp/pull/1365#discussion_r406098600) left that I took care of for you 👍 

Other than that, my only question is about the transient itself:
* The transient is set, but never seems to be retrieved - is that part going to be handled in another ticket?
* Similarly, there isn't anything preventing the same user from creating another ticket if they already have one - is that intentional? Not sure if they expire, but I would expect that if the transient has a 15min TTL then that same ticket should be used until it expires or is deleted.

Perhaps these were already discussed but I didn't see anything. Just trying to understand the flow 😄

**Comment by adamsilverstein** [comment]
> there was one minor fix left that I took care of for you 

Oh! Thank you.

> The transient is set, but never seems to be retrieved - is that part going to be handled in another ticket?

Right! After the user completes the Terms of Service, a GET request with the final created account details is made back to the passed `redirect_uri` (which in our case transits via the proxy). This is handled in https://github.com/google/site-kit-wp/issues/1213 where we will retrieve the transient and check the returned value with the stored value. This acts as a security measure similar to a nonce.

> Similarly, there isn't anything preventing the same user from creating another ticket if they already have one - is that intentional? Not sure if they expire, but I would expect that if the transient has a 15min TTL then that same ticket should be used until it expires or is deleted

Interesting question. I will do some follow up testing: my guess is that the `createAccoutTicket` API call will return the same id when called repeatedly. If not, we still want to reset the transient - we do this immediately before redirecting to the TOS and use it immediately upon return.

More details: https://developers.google.com/analytics/devguides/config/provisioning/v3/user-controlled#technical-tos-flow-response

**Comment by felixarntz** [comment]
@adamsilverstein @aaemnnosttv I don't think we need to add support for multiple account ticket IDs per user. When one is created, the feedback should be immediate. If another one is created while the old one still exists (highly unlikely with how the plugin works), it can just override that.

**Pull request merged**